### DISCUSSION TITLE
Cannot handle unknown node `raw` while migrating to MDX 2

### DISCUSSION THREAD

**Discussion by handotdev** [mdx-js/mdx#2117] [Q&A] [answered]
Hi everyone! I'm trying to migrate a legacy NextJS project into MDX 2. I ported everything to support ESM modules, and upgraded the following mdx dependencies from
```json
"@mdx-js/loader": "^1.6.21",
"@next/mdx": "^10.0.1",
```

to

```json
"@mdx-js/loader": "^2.1.3",
"@mdx-js/react": "^2.1.3",
"@next/mdx": "^12.2.5",
```

But am now presented with the following error. I've tried installing `rehype-raw` according to [this post](https://github.com/mdx-js/mdx/issues/1820) but it didn't solve it for me.

<img width="1023" alt="Screen Shot 2022-08-27 at 11 00 53 AM" src="https://user-images.githubusercontent.com/44352119/187042438-91c51b10-fed6-434f-8f44-8394058d28e9.png">

**Comment by wooorm** [comment, marked as answer]
See the previous discussion: https://github.com/mdx-js/mdx/discussions/2114.

**Reply by wooorm** [reply]
This is likely due to a certain plugin you are using. Which should be updated.
Or, you’re loading plain markdown. In which case, we don’t pass strings of HTML through to React (we can’t; that takes a ton of code. `rehype-raw` does that.)
